scipy/signal/_signaltools.py

```diff
@@ -1551,6 +1546,10 @@ def medfilt(volume, kernel_size=None):
 
     """
     volume = np.atleast_1d(volume)
+    if volume.dtype in [np.bool_, np.complex64, np.complex128, np.clongdouble,
+                        np.float16, np.object_, np.longdouble]:
+        raise ValueError(f"dtype={volume.dtype} is not supported by medfilt")
```

Comment:
I think `np.longdouble` here is incorrect, because on windows that just ends up being `float64` and then causes
```
FAILED scipy\signal\tests\test_signaltools.py::TestMedFilt::test_types[float64] - ValueError: dtype=float64 is not supported by medfilt
```

IMO it would be better to formulate this as:
```suggestion
    if volume.dtype not in <list_of_allowed_types>:
        raise ValueError(f"dtype={volume.dtype} is not supported by medfilt")
```